### PR TITLE
Fixed mdcNotifyProps super calls

### PR DESCRIPTION
**Pull request by cromefire**
Some super calls where forgotten to implement, so `mdcNotifyProps` didn't work.

Also they are now in `componentWillReceiveProps` instead of `componentWillUpdate`, I don't know the difference, but it seems right

### CODE REVIEW

**Comment by cromefire** [comment]
@prateekbh Could you please confirm this bug-fix?

File: ts/LinearProgress/index.tsx [hunk 1]
```diff
@@ -31,8 +31,8 @@ export class LinearProgress extends MaterialComponent<
     this.afterComponentDidMount();
   }
 
-  public componentWillUpdate(nextProps: ILinearProgressProps) {
-    super.componentWillUpdate(nextProps);
+  public componentWillReceiveProps(nextProps: ILinearProgressProps) {
```
Comment (prateekbh) [inline ts/LinearProgress/index.tsx]: add a todo to to switch away from `cWRP` as it is now deprecated

**Comment by cromefire** [comment]
Should we use componentWillUpdate instead?

File: ts/LinearProgress/index.tsx (hunk 1 above)
Comment (cromefire) [reply ts/LinearProgress/index.tsx]: 2.0

**Comment by cromefire** [comment]
Added an issue to remove all calls until `2.0`

**Comment by prateekbh** [comment]
All will* callbacks are deprecated

**Comment by cromefire** [comment]
And how are we supposed to handle that?

**Comment by cromefire** [comment]
So should we do everything in componentDid*?

**Comment by prateekbh** [comment]
getsnapshotbeforeupdate/ componentDidUpdate/useEffect(because hooks)

**Comment by cromefire** [comment]
Okay, I will look those up

**Pull request merged**